src/loaddb/load_session.hpp

```diff
@@ -155,6 +162,26 @@ namespace cubload
 
 } // namespace cubload
 
+namespace cubload
+{
+  // Template implementation
+  template<typename... Args>
+  void
+  session::append_log_msg (MSGCAT_LOADDB_MSG msg_id, Args &&... args)
+  {
+    if (get_args ().verbose)
+      {
+	std::string log_msg;
+	error_handler dummy_err (*this);
+	log_msg = dummy_err.format_log_msg (msg_id, std::forward<Args> (args)...);
```

Comment:
make format_log_msg static and avoid using dummy_err.

implementations of inline and template function/classes should be left at the end (in this case, after alias declaration).